sql/upgrade/sl3.0.sql

```diff
@@ -903,60 +910,71 @@ ALTER TABLE gl ENABLE TRIGGER gl_audit_trail;
+        duedate, invoice, ordnumber,
+        curr,
+        notes, quonumber, intnotes,
         shipvia, language_code, ponumber, shippingpoint,
-        on_hold, approved, reverse, terms, description)
+        on_hold, approved, reverse,
+        terms, description)
 SELECT
         customer.credit_id,
         (select entity_id from :slschema.employee WHERE id = ar.employee_id),
-        ar.id, invnumber, transdate, ar.taxincluded, amount, netamount,
-        CASE WHEN exchangerate IS NOT NULL THEN amount/exchangerate ELSE 0 END,
-        CASE WHEN exchangerate IS NOT NULL THEN netamount/exchangerate ELSE 0 END,
+        ar.id, invnumber, transdate, ar.taxincluded,
+        amount, netamount,
+        CASE WHEN exchangerate IS NOT NULL AND exchangerate <> 1 THEN amount/exchangerate ELSE 0 END,
+        CASE WHEN exchangerate IS NOT NULL AND exchangerate <> 1 THEN netamount/exchangerate ELSE 0 END,
 <?lsmb IF VERSION_COMPARE(lsmbversion,'1.6') < 0; ?>
         paid, datepaid,
 <?lsmb END; ?>
-        duedate, invoice, ordnumber, ar.curr, ar.notes, quonumber,
-        intnotes,
+        duedate, invoice, ordnumber,
+        CASE WHEN exchangerate IS NOT NULL THEN ar.curr ELSE NULL END,
+        ar.notes, quonumber, intnotes,
         shipvia, ar.language_code, ponumber, shippingpoint,
         onhold, approved, case when amount < 0 then true else false end,
         ar.terms, description
-FROM :slschema.ar JOIN :slschema.customer ON (ar.customer_id = customer.id) ;
+FROM :slschema.ar
+JOIN :slschema.customer ON (ar.customer_id = customer.id) ;
 
 ALTER TABLE ar ENABLE TRIGGER ar_audit_trail;
 
 ALTER TABLE ap DISABLE TRIGGER ap_audit_trail;
 
 insert into ap
-(entity_credit_account, person_id,
+        (entity_credit_account,
+        person_id,
         id, invnumber, transdate, taxincluded, amount_bc, netamount_bc,
-        amount_tc, netamount_tc,
+        amount_tc,
+        netamount_tc,
 <?lsmb IF VERSION_COMPARE(lsmbversion,'1.6') < 0; ?>
         paid, datepaid,
 <?lsmb END; ?>
-        duedate, invoice, ordnumber, curr, notes, quonumber, intnotes,
+        duedate, invoice, ordnumber,
+        curr,
+        notes, quonumber, intnotes,
         shipvia, language_code, ponumber, shippingpoint,
-        on_hold, approved, reverse, terms, description)
+        on_hold, approved, reverse,
+        terms, description)
 SELECT
         vendor.credit_id,
         (select entity_id from :slschema.employee
                 WHERE id = ap.employee_id),
         ap.id, invnumber, transdate, ap.taxincluded, amount, netamount,
-        CASE WHEN exchangerate IS NOT NULL THEN amount/exchangerate ELSE 0 END,
-        CASE WHEN exchangerate IS NOT NULL THEN netamount/exchangerate ELSE 0 END,
+        CASE WHEN exchangerate IS NOT NULL AND exchangerate <> 1 THEN amount/exchangerate ELSE 0 END,
+        CASE WHEN exchangerate IS NOT NULL AND exchangerate <> 1 THEN netamount/exchangerate ELSE 0 END,
```

Comment:
Should I conclude that you're setting `amount_tc` and `netamount_tc` to zero when the transaction is in the default currency? If that's the case, that's not how LedgerSMB's own schema migration should work: when the transaction is in the functional currency, the `netamount_bc == netamount_tc`.